lib/postgrex/pgpass.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Postgrex.Pgpass do
+  @moduledoc """
+  Parses .pgpass file to obtain database credentials
+  """
+
+  use Bitwise, only_operators: true
+
+  @doc """
+  Obtains the password given a keyword list containing hostname, database, port and optionally the username.
+  """
+  def password(opts) do
+    case pgpass_for(opts[:hostname], opts[:database], opts[:port], opts[:username], opts[:passfile]) do
+      [_h, _p, _d, _u, password] -> password
+      _ -> nil
+    end
+  end
+
+  defp pgpass_for(hostname, database, port, username, passfile) do
+    with {:ok, path} <- readable_pgpass_file?(passfile),
+         {:ok, contents} <- File.read(path),
+         entries when is_list(entries) <- parse(contents) do
+      entries
+      |> filter_hostname(hostname)
+      |> filter_port(port)
+      |> filter_database(database)
+      |> filter_username(username)
+      |> List.first
+    end
+  end
+
+  defp parse(contents) do
+    contents
+    |> String.split("\n")
+    |> Enum.reject(&(String.match?(&1,~r/^#/)))
+    |> Enum.map(&(String.split(&1, ":")))
```

Comment:
`Enum.map(&(String.split(&1, ":")))` is incorrect. The pgpassfile format can have `\`-escaped colons. `String.split/2` with the `parts` option set to 5 might be enough to ensure that the erroneous case isn't hit.

Sources: https://wiki.postgresql.org/wiki/Pgpass https://www.postgresql.org/docs/current/static/libpq-pgpass.html